### PR TITLE
plugin: unit tests.

### PR DESCRIPTION
**Pull request by swaptr**
this pr will add unit testing to the `cln_plugin` package.

Fixes https://github.com/dart-lightning/lndart.clightning/issues/56
Fixes https://github.com/dart-lightning/lndart.clightning/issues/41

### CODE REVIEW

File: packages/cln_plugin/test/cln_plugin_method_test.dart
Comment (vincenzopalazzo) [inline packages/cln_plugin/test/cln_plugin_method_test.dart:101]:
```suggestion
       expect(plugin.getManifest['subscriptions'], notNull);
       expect(plugin.getManifest['subscriptions']![0], "connect");
```

`expect(plugin.subscriptions['connect']!.method, 'connect');` is like `assert 1 + 1 == 2` :)

File: packages/cln_plugin/test/cln_plugin_method_test.dart
Comment (vincenzopalazzo) [inline packages/cln_plugin/test/cln_plugin_method_test.dart:118]: Very good here!

File: packages/cln_plugin/test/cln_plugin_method_test.dart
Comment (vincenzopalazzo) [inline packages/cln_plugin/test/cln_plugin_method_test.dart:124]: I don't think we can test this, or maybe yes

you can call `plugin.init(plugin, /*CORE LIGHTNING REQUEST*/)` you find the request on the cln docs :)

and after that check if the propriety in the plugin are in the correct place

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
Good starting point, I aded some suggestion

File: packages/cln_plugin/test/cln_plugin_method_test.dart
Comment (swaptr) [reply packages/cln_plugin/test/cln_plugin_method_test.dart:101]: oh no! I didnt have anything else to check here, hence this dumb mistake :D

File: packages/cln_plugin/test/cln_plugin_method_test.dart
Comment (swaptr) [inline packages/cln_plugin/test/cln_plugin_method_test.dart:157]: when I try to access properties like `options` and `configuration`, it shows them as empty.

Comment (vincenzopalazzo) [reply packages/cln_plugin/test/cln_plugin_method_test.dart:153]: The option inside the configuration is not the option that is registered inside the plugin.

You need to register the option `World` inside the plugin, or remove it from the configuration. 

We don't check this missing key in our code because core lightning send us the option that we specify in the manifest so this error should never happens.

Now the correct fix for this is the following one.

1. Add a test with the option registered in the plugin
2. Add a test without option inside the configuration
3. Make a sanity check that the key (optionName) exist inside the init method to avoid generic error like the one that you have in the test

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
Found a small bug in the test but after that this test is good to land

File: packages/cln_plugin/test/cln_plugin_method_test.dart
Comment (vincenzopalazzo) [inline packages/cln_plugin/test/cln_plugin_method_test.dart:161]:
```suggestion
      expect(plugin.options["greeting"]!.name.toString(), "greeting");
      expect(plugin.options["greeting"]!.value.toString(), "World");
```

you are checking that this is null before so you can add this assertion

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

**Review by vincenzopalazzo — APPROVED** [review]
Cool!

I added a sanity check to produce better error message https://github.com/dart-lightning/lndart.clightning/pull/61/commits/8f41b2cac710102c305a3733cd26602f5447ac5e and I check that this error message is throws in https://github.com/dart-lightning/lndart.clightning/pull/61/commits/06a7b3ed7b40f25c813ff55b8709e1056368298a

So let's wait for the CI and merge it!

Good job, next step hook?

**Pull request merged**
